src/Controllers/Interface/Forms/Observation/ObservationController.php

```diff
@@ -0,0 +1,658 @@
+class ObservationController
+{
+    private ObservationService $observationService;
+
+    private FormService $formService;
+
+    public function __construct(?ObservationService $observationService = null, ?FormService $formService = null)
+    {
+        $this->observationService = $observationService ?? new ObservationService();
+        $this->formService = $formService ?? new FormService();
+    }
+
+    /**
+     * Handle new/edit observation form display
+     *
+     * @param Request $request
+     * @return Response
+     */
+    public function new(Request $request): Response
+    {
+        $formId = $request->query->getInt('id');
+        $observations = [];
+
+        if ($formId) {
+            $observations = $this->observationService->getObservationsByFormId(
+                $formId,
+                $_SESSION["pid"],
+                $_SESSION["encounter"]
+            );
+        }
+
+        $reasonCodeStatii = ReasonStatusCodes::getCodesWithDescriptions();
+        $reasonCodeStatii[ReasonStatusCodes::NONE]['description'] = xl("Select a status code");
+        $obTypes = $this->observationService->getObservationTypes();
+
+        ob_start();
+        $this->renderNewForm($formId, $observations, $reasonCodeStatii, $obTypes);
+        $content = ob_get_clean();
+        return $this->createResponse($content);
+    }
+
+    /**
+     * Handle observation form save
+     *
+     * @param Request $request
+     * @return Response
+     */
+    public function save(Request $request): Response
+    {
+        global $encounter;
+
+        $postData = $request->getPayload()->all();
+
+        // CSRF verification
+        if (!CsrfUtils::verifyCsrfToken($postData["csrf_token_form"])) {
+            (new SystemLogger())->errorLogCaller("OpenEMR CSRF token authentication error on observation form save");
+            return $this->createResponse("Authentication Error", Response::HTTP_UNAUTHORIZED);
+        }
+
+        if (!$encounter) {
+            (new SystemLogger())->errorLogCaller("Internal error: we do not seem to be in an encounter on observation form save");
+            return $this->createResponse(xlt("Internal error: we do not seem to be in an encounter!"), Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+
+        $formId = $request->query->getInt('id');
+        $newId = $this->processFormSave($formId, $postData);
+        return $this->createResponse($this->getFormJumpHtml("Redirecting...."));
+    }
+
+    private function getFormJumpHtml($title): string
+    {
+        ob_start();
+        ?>
+        <html>
+        <head>
+            <?php Header::setupHeader(); ?>
+            <title><?php echo text($title); ?></title>
+        </head>
+        <body style="background: <?php echo attr_url($GLOBALS['backpic'])?>; margin: 10px;">
+        <script>
+        parent.closeTab(window.name, true);
+        </script>
+        </body>
+        </html>
```

Comment:
Is this "view" code in the controller?

Reply:
They'll be converted to twig files on the next iteration.